src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustIndividualValidator.java

```diff
@@ -177,9 +175,20 @@ private boolean validateServiceAddressSameAsUsualResidentialAddress(Boolean same
         return UtilsValidators.isNotNull(same, qualifiedFieldName, errors, loggingContext);
     }
 
-    private void validateSameAsAddress(TrustIndividualDto trustIndividualDto, Errors errors, String loggingContext) {
-        boolean isSameAddressFlagValid = validateServiceAddressSameAsUsualResidentialAddress(
-                trustIndividualDto.getServiceAddressSameAsUsualResidentialAddress(), errors, loggingContext);
+    private void validateSameAsAddress(TrustIndividualDto trustIndividualDto, Errors errors, String loggingContext, boolean isForUpdateOrRemove) {
+        // Default value to true and if Dto is for a Registration, validate it properly.
+        // If Dto is for an Update or Remove then leave as true - this is because some data in mongo
+        // might already have been stored with a null value for Update/Remove before this validation was changed.
+        // If we try and refund a record with a null in the 'sameAs' field we don't want validation to now fail as
```

Comment:
My concerns around whether this is really an issue (for refunds) has re-surfaced, perhaps because of my time off...

Do we really need to be relaxing validation in a number of places to deal with this? Has this been proven via a test of the process with the application or is it a hunch/worry?

Reply:
@pvoddenCH did a test in CiDev and rejected an OE01. I changed the incorporation country to be a blank string in Mongo after it had arrived in CHIPS and before rejection - this would normally cause a validation error if `validation-status` end-point was called. The refund e-mail was still sent though and there were no errors in the API logs. The only activity in API log when refund was actioned seemed to be a call in (presumably from Transactions API) to get the costs.

Log extract:

2024-09-03T12:17:25.989Z
{"created":"2024-09-03T12:17:25.989Z","event":"debug","namespace":"overseas-entities-api","context":"6yKKfFaAA1gaFyh1H0O2TmTr_4wv","data":{"transaction_id":"081144-505917-253652","message":"UserAuthenticationInterceptor skipping token permission checks for api key request"}}
2024-09-03T12:17:25.989Z
{"created":"2024-09-03T12:17:25.989Z","event":"debug","namespace":"class uk.gov.companieshouse.api.interceptor.InternalUserInterceptor","context":"6yKKfFaAA1gaFyh1H0O2TmTr_4wv","data":{"message":"authorised as api key (internal user)"}}
2024-09-03T12:17:26.014Z
{"created":"2024-09-03T12:17:25.989Z","event":"debug","namespace":"overseas-entities-api","context":"6yKKfFaAA1gaFyh1H0O2TmTr_4wv","data":{"transaction_id":"081144-505917-253652","message":"Getting transaction for request."}}
2024-09-03T12:17:26.015Z
{"created":"2024-09-03T12:17:26.014Z","event":"debug","namespace":"overseas-entities-api","context":"6yKKfFaAA1gaFyh1H0O2TmTr_4wv","data":{"transaction_id":"081144-505917-253652","message":"Transaction retrieved."}}
2024-09-03T12:17:26.020Z
{"created":"2024-09-03T12:17:26.014Z","event":"info","namespace":"overseas-entities-api","context":"6yKKfFaAA1gaFyh1H0O2TmTr_4wv","data":{"transaction_id":"081144-505917-253652","message":"Calling CostsService to retrieve costs","overseas_entity_id":"66d6fc0f75706c7b6493f965"}}
2024-09-03T12:17:26.022Z
{"created":"2024-09-03T12:17:26.020Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Schema version is VERSION_3_2"}}
2024-09-03T12:17:26.022Z
{"created":"2024-09-03T12:17:26.022Z","event":"info","namespace":"overseas-entities-api","data":{"message":"66d6fc0f75706c7b6493f965: Overseas Entities Submission found. Schema version 3.2"}}
2024-09-03T12:17:26.026Z
{"created":"2024-09-03T12:17:26.022Z","event":"info","namespace":"overseas-entities-api","context":"6yKKfFaAA1gaFyh1H0O2TmTr_4wv","data":{"message":"Registration submission found"}}
2024-09-03T12:17:26.031Z
{"created":"2024-09-03T12:17:26.026Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Schema version is VERSION_3_2"}}
2024-09-03T12:17:26.031Z
{"created":"2024-09-03T12:17:26.031Z","event":"info","namespace":"overseas-entities-api","data":{"message":"66d6fc0f75706c7b6493f965: Overseas Entities Submission found. Schema version 3.2"}}
2024-09-03T12:17:26.035Z
{"created":"2024-09-03T12:17:26.031Z","event":"info","namespace":"overseas-entities-api","context":"6yKKfFaAA1gaFyh1H0O2TmTr_4wv","data":{"message":"Registration submission found"}}
2024-09-03T12:17:26.070Z
{"created":"2024-09-03T12:17:26.033Z","event":"info","namespace":"overseas-entities-api","context":"6yKKfFaAA1gaFyh1H0O2TmTr_4wv","data":{"message":"End of request. Method: GET Path: /transactions/{transaction_id}/overseas-entity/{overseas_entity_id}/costs Duration: 45ms Status: 200"}}
2024-09-03T12:17:26.070Z
{"created":"2024-09-03T12:17:26.069Z","event":"info","namespace":"overseas-entities-api","context":"8Rk_aXnYH7DnBLqDAFICbRZaSD4_","data":{"message":"Start of request. Method: GET Path: /transactions/{transaction_id}/overseas-entity/{overseas_entity_id}/costs"}}
2024-09-03T12:17:26.070Z
{"created":"2024-09-03T12:17:26.069Z","event":"debug","namespace":"class uk.gov.companieshouse.api.interceptor.TokenPermissionsInterceptor","context":"8Rk_aXnYH7DnBLqDAFICbRZaSD4_","data":{"message":"Create TokenPermissions and store it in request"}}
2024-09-03T12:17:26.070Z
{"created":"2024-09-03T12:17:26.070Z","event":"debug","namespace":"class uk.gov.companieshouse.api.util.security.TokenPermissionsImpl","context":"8Rk_aXnYH7DnBLqDAFICbRZaSD4_","data":{"Token permissions":{},"message":"Parsed ERIC token permissions"}}
2024-09-03T12:17:26.070Z
{"created":"2024-09-03T12:17:26.070Z","event":"debug","namespace":"overseas-entities-api","context":"8Rk_aXnYH7DnBLqDAFICbRZaSD4_","data":{"transaction_id":"081144-505917-253652","message":"UserAuthenticationInterceptor skipping token permission checks for api key request"}}
2024-09-03T12:17:26.070Z
{"created":"2024-09-03T12:17:26.070Z","event":"debug","namespace":"class uk.gov.companieshouse.api.interceptor.InternalUserInterceptor","context":"8Rk_aXnYH7DnBLqDAFICbRZaSD4_","data":{"message":"authorised as api key (internal user)"}}
2024-09-03T12:17:26.083Z
{"created":"2024-09-03T12:17:26.070Z","event":"debug","namespace":"overseas-entities-api","context":"8Rk_aXnYH7DnBLqDAFICbRZaSD4_","data":{"transaction_id":"081144-505917-253652","message":"Getting transaction for request."}}
2024-09-03T12:17:26.083Z
{"created":"2024-09-03T12:17:26.083Z","event":"debug","namespace":"overseas-entities-api","context":"8Rk_aXnYH7DnBLqDAFICbRZaSD4_","data":{"transaction_id":"081144-505917-253652","message":"Transaction retrieved."}}
2024-09-03T12:17:26.087Z
{"created":"2024-09-03T12:17:26.083Z","event":"info","namespace":"overseas-entities-api","context":"8Rk_aXnYH7DnBLqDAFICbRZaSD4_","data":{"transaction_id":"081144-505917-253652","message":"Calling CostsService to retrieve costs","overseas_entity_id":"66d6fc0f75706c7b6493f965"}}
2024-09-03T12:17:26.088Z

Reply:
Just tested resubmission via the FESS admin function and that also seems to be fine when the Mongo data contains invalid data. It would appear that the `validation-status` end-point isn't hit but rather the `/costs` and `/filings` end-points are invoked. The filing was re-processed by chips-filing-consumer with no errors in any logs.

OE API log extract:

{"created":"2024-09-04T10:16:21.404Z","event":"debug","namespace":"class uk.gov.companieshouse.api.interceptor.InternalUserInterceptor","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"message":"authorised as api key (internal user)"}}
{"created":"2024-09-04T10:16:21.404Z","event":"debug","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"transaction_id":"020796-687117-254447","message":"Getting transaction for request."}}
{"created":"2024-09-04T10:16:21.432Z","event":"debug","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"transaction_id":"020796-687117-254447","message":"Transaction retrieved."}}
{"created":"2024-09-04T10:16:21.433Z","event":"debug","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"message":"Called preHandle(...)"}}
{"created":"2024-09-04T10:16:21.433Z","event":"info","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"transaction_id":"020796-687117-254447","message":"Transaction is closed - filing allowed"}}
{"created":"2024-09-04T10:16:21.435Z","event":"info","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"transaction_id":"020796-687117-254447","message":"Calling service to retrieve filing"}}
{"created":"2024-09-04T10:16:21.435Z","event":"info","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"transaction_id":"020796-687117-254447","message":"Transaction id is 020796-687117-254447"}}
{"created":"2024-09-04T10:16:21.439Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Schema version is VERSION_3_2"}}
{"created":"2024-09-04T10:16:21.441Z","event":"info","namespace":"overseas-entities-api","data":{"message":"66d832866aca907c9797d08f: Overseas Entities Submission found. Schema version 3.2"}}
{"created":"2024-09-04T10:16:21.441Z","event":"debug","namespace":"overseas-entities-api","data":{"transaction_id":"020796-687117-254447","message":"Value of 'isNoChange' flag for Update is :false","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.441Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Retrieving Company Profile for Company Number OE001022","company_number":"OE001022"}}
{"created":"2024-09-04T10:16:21.508Z","event":"info","namespace":"overseas-entities-api","data":{"message":"No overseas entity data for Company Number OE001022"}}
{"created":"2024-09-04T10:16:21.508Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Retrieving Beneficial Owners for Company Number OE001022","company_number":"OE001022"}}
{"created":"2024-09-04T10:16:21.540Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Retrieved 2 Beneficial Owners for Company Number OE001022"}}
{"created":"2024-09-04T10:16:21.540Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Retrieving PSCs data for Company Number OE001022","company_number":"OE001022"}}
{"created":"2024-09-04T10:16:21.593Z","event":"info","namespace":"overseas-entities-api","data":{"message":"No Managing Officers data found for Company Number OE001022","company_number":"OE001022"}}
{"created":"2024-09-04T10:16:21.593Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Retrieving Officers data for Company Number OE001022","company_number":"OE001022"}}
{"created":"2024-09-04T10:16:21.616Z","event":"info","namespace":"overseas-entities-api","context":"PublicPrivateDataCombiner","data":{"message":"{\"Managing Officer Data\": [],\"Beneficial Owner Data\": [{\"Public Hashed ID\": \"XNsrmgVpfhT4sXdTlVqsdDXoCHA\", \"Private ID\": \"null\"}, {\"Public Hashed ID\": \"zWGKy0aCHpuDaTWHfRrLNR3Pkr4\", \"Private ID\": \"null\"}, {\"Public Hashed ID\": \"null\", \"Private ID\": \"4000002087\"}, {\"Public Hashed ID\": \"null\", \"Private ID\": \"4000002088\"}],\"Overseas Entity Data\": {\"Public Entity Number\": \"OE001022\", \"Private Email\": \"null\"}}"}}
{"created":"2024-09-04T10:16:21.616Z","event":"error","namespace":"overseas-entities-api","context":"BeneficialOwnerChangeService","data":{"transaction_id":"020796-687117-254447","message":"No private data found for beneficial owner - changes cannot be created","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.616Z","event":"error","namespace":"overseas-entities-api","context":"BeneficialOwnerChangeService","data":{"transaction_id":"020796-687117-254447","message":"No private data found for beneficial owner - changes cannot be created","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.617Z","event":"debug","namespace":"overseas-entities-api","data":{"transaction_id":"020796-687117-254447","message":"Value of 'changes' is :[uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.PrincipalAddressChange@33adc8cc, uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.CorrespondenceAddressChange@607b2900, uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.CompanyIdentificationChange@3219abd8, uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.EntityEmailAddressChange@3df8c0f0]","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.617Z","event":"debug","namespace":"overseas-entities-api","data":{"transaction_id":"020796-687117-254447","message":"Updates have been collected","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.617Z","event":"debug","namespace":"overseas-entities-api","data":{"transaction_id":"020796-687117-254447","message":"Adding Agent due diligence data to filing","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.617Z","event":"debug","namespace":"overseas-entities-api","data":{"transaction_id":"020796-687117-254447","message":"Adding 'changes' data to filing","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.617Z","event":"debug","namespace":"overseas-entities-api","data":{"transaction_id":"020796-687117-254447","message":"Update/Remove submission data has been set on filing: {anyBOsOrMOsAddedOrCeased=false, beneficialOwnerStatement=ALL_IDENTIFIED_ALL_DETAILS, presenter=uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.Presenter@60ce942b, additions=[], cessations=[], trusts=[uk.gov.companieshouse.overseasentitiesapi.model.dto.trust.TrustDataDto@5944821b], entityName=PV 1 IND 1 CORP WITH TRUSTS 23/02, entityNumber=OE001022, userSubmission=uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto@267cb580, changes=[uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.PrincipalAddressChange@33adc8cc, uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.CorrespondenceAddressChange@607b2900, uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.CompanyIdentificationChange@3219abd8, uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.EntityEmailAddressChange@3df8c0f0], dueDiligence=uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.DueDiligence@5df81eff}","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.617Z","event":"debug","namespace":"overseas-entities-api","data":{"transaction_id":"020796-687117-254447","message":"Update specific submission data has been set on filing: {beneficialOwnerStatement=ALL_IDENTIFIED_ALL_DETAILS, presenter=uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.Presenter@60ce942b, additions=[], entityNumber=OE001022, changes=[uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.PrincipalAddressChange@33adc8cc, uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.CorrespondenceAddressChange@607b2900, uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.CompanyIdentificationChange@3219abd8, uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.EntityEmailAddressChange@3df8c0f0], dueDiligence=uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.DueDiligence@5df81eff, filingForDate=uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.FilingForDate@4983a6a7, trusteeInvolvedRelevantPeriodStatement=null, anyBOsOrMOsAddedOrCeased=false, cessations=[], trusts=[uk.gov.companieshouse.overseasentitiesapi.model.dto.trust.TrustDataDto@5944821b], entityName=PV 1 IND 1 CORP WITH TRUSTS 23/02, changeBORelevantPeriodStatement=null, changeBeneficiaryRelevantPeriodStatement=null, userSubmission=uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto@267cb580}","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.650Z","event":"info","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"message":"Start of request. Method: GET Path: /transactions/{transaction_id}/overseas-entity/{overseas_entity_id}/costs"}}
{"created":"2024-09-04T10:16:21.650Z","event":"debug","namespace":"class uk.gov.companieshouse.api.interceptor.TokenPermissionsInterceptor","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"message":"Create TokenPermissions and store it in request"}}
{"created":"2024-09-04T10:16:21.650Z","event":"debug","namespace":"class uk.gov.companieshouse.api.util.security.TokenPermissionsImpl","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"Token permissions":{},"message":"Parsed ERIC token permissions"}}
{"created":"2024-09-04T10:16:21.650Z","event":"debug","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"transaction_id":"020796-687117-254447","message":"UserAuthenticationInterceptor skipping token permission checks for api key request"}}
{"created":"2024-09-04T10:16:21.650Z","event":"debug","namespace":"class uk.gov.companieshouse.api.interceptor.InternalUserInterceptor","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"message":"authorised as api key (internal user)"}}
{"created":"2024-09-04T10:16:21.650Z","event":"debug","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"transaction_id":"020796-687117-254447","message":"Getting transaction for request."}}
{"created":"2024-09-04T10:16:21.669Z","event":"debug","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"transaction_id":"020796-687117-254447","message":"Transaction retrieved."}}
{"created":"2024-09-04T10:16:21.670Z","event":"info","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"transaction_id":"020796-687117-254447","message":"Calling CostsService to retrieve costs","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.672Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Schema version is VERSION_3_2"}}
{"created":"2024-09-04T10:16:21.673Z","event":"info","namespace":"overseas-entities-api","data":{"message":"66d832866aca907c9797d08f: Overseas Entities Submission found. Schema version 3.2"}}
{"created":"2024-09-04T10:16:21.673Z","event":"info","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"message":"Update submission with overseas entity number OE001022 found"}}
{"created":"2024-09-04T10:16:21.676Z","event":"info","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"message":"End of request. Method: GET Path: /transactions/{transaction_id}/overseas-entity/{overseas_entity_id}/costs Duration: 26ms Status: 200"}}
{"created":"2024-09-04T10:16:21.712Z","event":"info","namespace":"overseas-entities-api","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"message":"Start of request. Method: GET Path: /transactions/{transaction_id}/overseas-entity/{overseas_entity_id}/costs"}}
{"created":"2024-09-04T10:16:21.713Z","event":"debug","namespace":"class uk.gov.companieshouse.api.interceptor.TokenPermissionsInterceptor","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"message":"Create TokenPermissions and store it in request"}}
{"created":"2024-09-04T10:16:21.713Z","event":"debug","namespace":"class uk.gov.companieshouse.api.util.security.TokenPermissionsImpl","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"Token permissions":{},"message":"Parsed ERIC token permissions"}}
{"created":"2024-09-04T10:16:21.713Z","event":"debug","namespace":"overseas-entities-api","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"transaction_id":"020796-687117-254447","message":"UserAuthenticationInterceptor skipping token permission checks for api key request"}}
{"created":"2024-09-04T10:16:21.713Z","event":"debug","namespace":"class uk.gov.companieshouse.api.interceptor.InternalUserInterceptor","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"message":"authorised as api key (internal user)"}}
{"created":"2024-09-04T10:16:21.713Z","event":"debug","namespace":"overseas-entities-api","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"transaction_id":"020796-687117-254447","message":"Getting transaction for request."}}
{"created":"2024-09-04T10:16:21.729Z","event":"debug","namespace":"overseas-entities-api","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"transaction_id":"020796-687117-254447","message":"Transaction retrieved."}}
{"created":"2024-09-04T10:16:21.730Z","event":"info","namespace":"overseas-entities-api","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"transaction_id":"020796-687117-254447","message":"Calling CostsService to retrieve costs","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.732Z","event":"info","namespace":"overseas-entities-api","data":{"message":"Schema version is VERSION_3_2"}}
{"created":"2024-09-04T10:16:21.734Z","event":"info","namespace":"overseas-entities-api","data":{"message":"66d832866aca907c9797d08f: Overseas Entities Submission found. Schema version 3.2"}}
{"created":"2024-09-04T10:16:21.734Z","event":"info","namespace":"overseas-entities-api","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"message":"Update submission with overseas entity number OE001022 found"}}
{"created":"2024-09-04T10:16:21.735Z","event":"info","namespace":"overseas-entities-api","context":"VOluIwqlncsvuA8O2-PIVww4wbj_","data":{"message":"End of request. Method: GET Path: /transactions/{transaction_id}/overseas-entity/{overseas_entity_id}/costs Duration: 23ms Status: 200"}}
{"created":"2024-09-04T10:16:21.751Z","event":"debug","namespace":"overseas-entities-api","data":{"transaction_id":"020796-687117-254447","message":"Payment data has been set on filing","overseas_entity_id":"66d832866aca907c9797d08f"}}
{"created":"2024-09-04T10:16:21.759Z","event":"info","namespace":"overseas-entities-api","context":"lOS7dWzF6CZ5-52Jfq_TqrLwo0nz","data":{"message":"End of request. Method: GET Path: /private/transactions/{transaction_id}/overseas-entity/{overseas_entity_id}/filings Duration: 356ms Status: 200"}}
{"data":{"message":"Exception in monitor thread while connecting to server mongo:27017"},"created":"2024-09-04T10:29:53.347Z","event":"info"}
{"data":{"message":"Exception in monitor thread while connecting to server mongo:27017"},"created":"2024-09-04T10:29:53.420Z","event":"info"}
{"data":{"message":"Monitor thread successfully connected to server with description ServerDescription{address=mongo:27017, type=REPLICA_SET_PRIMARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=24033375, setName='tilt-rs', canonicalAddress=mongo:27017, hosts=[mongo:27017], passives=[], arbiters=[], primary='mongo:27017', tagSet=TagSet{[]}, electionId=7fffffff0000000000000001, setVersion=1, topologyVersion=TopologyVersion{processId=66d82efb077aea350c502566, counter=6}, lastWriteDate=Wed Sep 04 10:29:47 UTC 2024, lastUpdateTimeNanos=29992758719340}"},"created":"2024-09-04T10:29:53.693Z","event":"info"}

Reply:
agreed, will undo the change to allow null

Reply:
reverted the 'allow null' change.